### DISCUSSION TITLE
Flaky test detection and automatic quarantining in CI

### DISCUSSION THREAD

**Discussion by kapoorlakshya** [MobileNativeFoundation/discussions#186] [General]
Hey all, curious what you all are doing or using (bought or built) to deal with detection of flaky tests and automatically quarantining them from CI runs.

It appears that the bigger orgs prefer to build their own and there are a bunch of paid services for this now. I also found a few open source tools to do this, but curious what's working well for mobile teams here.

**Relevant Links**
https://slack.engineering/handling-flaky-tests-at-scale-auto-detection-suppression/
https://engineering.fb.com/2020/12/10/developer-tools/probabilistic-flakiness/
https://www.cs.cornell.edu/~legunsen/pubs/BellETAL18DeFlaker.pdf
https://github.com/F-Secure/flaky-tests-detection
https://testing.googleblog.com/2016/05/flaky-tests-at-google-and-how-we.html
https://buildpulse.io/
https://github.blog/2020-12-16-reducing-flaky-builds-by-18x/

**Comment by keith** [comment]
I think this is a great list of links, I don't have much to add except that we're also hoping buildkite's (our CI provider) product expands to support this https://buildkite.com/test-analytics today we rerun things automatically but don't do anything smart like this yet
